vertx-oracle-client/src/main/java/io/vertx/oracleclient/OracleConnection.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2011-2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.oracleclient;
+
+import io.vertx.codegen.annotations.Fluent;
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.oracleclient.impl.OracleConnectionImpl;
+import io.vertx.sqlclient.PreparedStatement;
+import io.vertx.sqlclient.SqlConnection;
+
+import static io.vertx.oracleclient.OracleConnectOptions.fromUri;
+
+/**
+ * A connection to Oracle Database.
+ */
+@VertxGen
+public interface OracleConnection extends SqlConnection {
+
+  /**
+   * Create a connection to Oracle with the given {@code connectOptions}.
+   *
+   * @param vertx the vertx instance
+   * @param connectOptions the options for the connection
+   * @param handler the handler called with the connection or the failure
+   */
+  static void connect(Vertx vertx, OracleConnectOptions connectOptions, Handler<AsyncResult<OracleConnection>> handler) {
+    Future<OracleConnection> fut = OracleConnectionImpl.connect(vertx, connectOptions);
+    if (handler != null) {
```

Comment:
so what happens when the handler is null? My guess is a connection is eventually opened, and it's leaked?

In that case, wouldn't it be better for this code to throw an exception before attempting to connect?